File: backend/simple/payment/src/proto-services/health.rb
require_relative '../../pb/health_pb'
require_relative '../../pb/health_services_pb'

class HealthServicer < Grpc::Health::V1::Health::Service
    def check(health_check_request, _unused_call)
        Grpc::Health::V1::HealthCheckResponse.new(status: :SERVING)
    end

    def watch(health_check_request, _call)
        return enum_for(:watch, health_check_request, _call) unless block_given?
        loop do
            yield Grpc::Health::V1::HealthCheckResponse.new(status: :SERVING)
            sleep 1
        end
    end
end

File: backend/simple/user-storage/src/proto-services/health.ts
import * as grpc from '@grpc/grpc-js';

import * as health_pb from "../../pb/health"

class Health extends health_pb.grpc.health.v1.UnimplementedHealthService {
    Check(call: grpc.ServerUnaryCall<health_pb.grpc.health.v1.HealthCheckRequest, health_pb.grpc.health.v1.HealthCheckResponse>, callback: grpc.sendUnaryData<health_pb.grpc.health.v1.HealthCheckResponse>): void {
        const response = new health_pb.grpc.health.v1.HealthCheckResponse({
            status: health_pb.grpc.health.v1.HealthCheckResponse.ServingStatus.SERVING,
        });

        callback(null, response);
    }

    Watch(call: grpc.ServerWritableStream<health_pb.grpc.health.v1.HealthCheckRequest, health_pb.grpc.health.v1.HealthCheckResponse>): void {
        const response = new health_pb.grpc.health.v1.HealthCheckResponse({
            status: health_pb.grpc.health.v1.HealthCheckResponse.ServingStatus.SERVING,
        });

        call.write(response);
    }
}

export default Health;